rust/common-runtime/src/runtime.rs

```diff
@@ -53,15 +68,25 @@ impl InputOutput for RuntimeIo {
         }
     }
 
-    fn output(&self) -> &BTreeMap<String, Value> {
+    fn input(&self) -> &DataIo {
+        &self.input
+    }
+
+    fn output(&self) -> &DataIo {
         &self.output
     }
 
-    fn output_shape(&self) -> &OutputShape {
+    fn output_shape(&self) -> &IoShape {
         &self.output_shape
     }
 }
 
+struct CachedModule {
```

Comment:
:pray: 

FWIW this is more correctly called a `LiveModule`. It's not retained for caching purposes; it's retained because its lifetime hasn't ended yet.